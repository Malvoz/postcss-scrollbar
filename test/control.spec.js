import fs from 'fs';
import path from 'path';
import postcss from 'postcss';
import plugin from '../src';

const pluginName = require('../package.json').name;

const read = name =>
  fs.readFileSync(path.join(__dirname, 'fixture', name), 'utf8');

const input = read('thin/input.css');

describe('control: ', () => {
  test('no options', () =>
    postcss([plugin])
      .process(input)
      .then(result => {
        expect(result.css).toMatchSnapshot();
      }));

  test('with options', () =>
    postcss([plugin({})])
      .process(input)
      .then(result => {
        expect(result.css).toMatchSnapshot();
      }));

  test('PostCSS legacy API', () => {
    const result = postcss([plugin.postcss]).process(input).css;
    expect(result).toMatchSnapshot();
  });

  test('PostCSS API', async () => {
    const processor = postcss();
    processor.use(plugin);

    const result = await processor.process(input);

    expect(result.css).toMatchSnapshot();
    expect(processor.plugins[0].postcssPlugin).toBe(pluginName);
    expect(processor.plugins[0].postcssVersion).toBeTruthy();
  });
});
