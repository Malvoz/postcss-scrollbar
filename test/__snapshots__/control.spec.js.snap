// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`control:  PostCSS API 1`] = `
".scrollable::-webkit-scrollbar-thumb {
  background-color: rebeccapurple;
}
.scrollable::-webkit-scrollbar-track {
  background-color: green;
}
.scrollable::-webkit-scrollbar {
  width: 0.5rem;
}
.scrollable {
  scrollbar-color: rebeccapurple green;
  scrollbar-width: thin;
}
"
`;

exports[`control:  PostCSS legacy API 1`] = `
".scrollable::-webkit-scrollbar-thumb {
  background-color: rebeccapurple;
}
.scrollable::-webkit-scrollbar-track {
  background-color: green;
}
.scrollable::-webkit-scrollbar {
  width: 0.5rem;
}
.scrollable {
  scrollbar-color: rebeccapurple green;
  scrollbar-width: thin;
}
"
`;

exports[`control:  no options 1`] = `
".scrollable::-webkit-scrollbar-thumb {
  background-color: rebeccapurple;
}
.scrollable::-webkit-scrollbar-track {
  background-color: green;
}
.scrollable::-webkit-scrollbar {
  width: 0.5rem;
}
.scrollable {
  scrollbar-color: rebeccapurple green;
  scrollbar-width: thin;
}
"
`;

exports[`control:  with options 1`] = `
".scrollable::-webkit-scrollbar-thumb {
  background-color: rebeccapurple;
}
.scrollable::-webkit-scrollbar-track {
  background-color: green;
}
.scrollable::-webkit-scrollbar {
  width: 0.5rem;
}
.scrollable {
  scrollbar-color: rebeccapurple green;
  scrollbar-width: thin;
}
"
`;
